lib/nfc/nfc_worker.c

```diff
@@ -453,6 +453,36 @@ void nfc_worker_emulate_mf_ultralight(NfcWorker* nfc_worker) {
     }
 }
 
+void nfc_worker_mf_classic_key_attack(NfcWorker* nfc_worker, uint64_t key, FuriHalNfcTxRxContext* tx_rx) {
+    furi_assert(nfc_worker);
+
+    MfClassicData* data = &nfc_worker->dev_data->mf_classic_data;
+    uint32_t total_sectors = mf_classic_get_total_sectors_num(data->type);
+
+    // Check every sector's A and B keys with the given key
+    for(size_t i = 0; i < total_sectors; i++) {
```

Comment:
Let's pass start sector to nfc_worker_mf_classic_key_attack() arguments. We don't need to try found key starting from 0 sector each time

Reply:
Yep, sound good!